gemstone/src/swapper/chainflip/broker/client.rs

```diff
@@ -0,0 +1,90 @@
+use super::{
+    model::{ChainflipAsset, DcaParameters, DepositAddressResponse, RefundParameters, VaultSwapExtraParams, VaultSwapResponse},
+    CahinflipIngressEgress, ChainflipEnvironment,
+};
+use crate::{
+    network::{jsonrpc::JsonRpcClient, AlienProvider},
+    swapper::SwapperError,
+};
+use serde_json::json;
+use std::sync::Arc;
+
+const CHAINFLIP_BROKER_URL: &str = "https://chainflip-broker.io";
+const CHAINFLIP_BROKER_KEY: &str = "ed08651813cc4d4798bf9b953b5d33fb";
+
+#[derive(Debug)]
+pub struct BrokerClient {
+    client: JsonRpcClient,
+}
+
+impl BrokerClient {
+    pub fn new(provider: Arc<dyn AlienProvider>) -> Self {
+        Self {
+            client: JsonRpcClient::new(provider.clone(), format!("{}/rpc/{}", CHAINFLIP_BROKER_URL, CHAINFLIP_BROKER_KEY)),
+        }
+    }
+
+    pub async fn get_swap_limits(&self) -> Result<CahinflipIngressEgress, SwapperError> {
+        self.client
+            .call_method_with_param("cf_environment", json!([]), Some(60 * 60 * 24 * 30))
+            .await
+            .map_err(SwapperError::from)
+            .map(|x| x.take().map_err(SwapperError::from))?
+            .map(|x: ChainflipEnvironment| x.ingress_egress)
+    }
+
+    pub async fn get_deposit_address(
+        &self,
+        src_asset: ChainflipAsset,
+        dst_asset: ChainflipAsset,
+        dst_address: String,
+        broker_commission_bps: u32,
+        refund_params: Option<RefundParameters>,
+        dca_params: Option<DcaParameters>,
+    ) -> Result<DepositAddressResponse, SwapperError> {
+        let params = json!([
+            src_asset,
+            dst_asset,
+            dst_address,
+            broker_commission_bps,
+            null, // channel_metadata
+            null, // boost_fee
```

Comment:
Check if source asset is BTC, and decide to set it to `5` if the user wants it (or default true). This would make a BTC -> .... swap start after 1 btc confirmation instead of 3 (dont forget to also pass it in the quote request)

Reply:
Thanks, added